src/scene/renderer/render-pass-cookie-renderer.js

```diff
@@ -0,0 +1,161 @@
+import { Vec4 } from '../../core/math/vec4.js';
+import { Mat4 } from '../../core/math/mat4.js';
+import { CULLFACE_NONE } from '../../platform/graphics/constants.js';
+import { DebugGraphics } from '../../platform/graphics/debug-graphics.js';
+import { LIGHTTYPE_OMNI } from '../constants.js';
+import { createShaderFromCode } from '../shader-lib/utils.js';
+import { LightCamera } from './light-camera.js';
+import { BlendState } from '../../platform/graphics/blend-state.js';
+import { QuadRender } from '../graphics/quad-render.js';
+import { DepthState } from '../../platform/graphics/depth-state.js';
+import { RenderPass } from "../../platform/graphics/render-pass.js";
+
+const textureBlitVertexShader = `
+    attribute vec2 vertex_position;
+    varying vec2 uv0;
+    void main(void) {
+        gl_Position = vec4(vertex_position, 0.5, 1.0);
+        uv0 = vertex_position.xy * 0.5 + 0.5;
+        #ifndef WEBGPU
+            uv0.y = 1.0 - uv0.y;
+        #endif
+    }`;
+
+const textureBlitFragmentShader = `
+    varying vec2 uv0;
+    uniform sampler2D blitTexture;
+    void main(void) {
+        gl_FragColor = texture2D(blitTexture, uv0);
+    }`;
+
+// shader runs for each face, with inViewProj matrix representing a face camera
+const textureCubeBlitFragmentShader = `
+    varying vec2 uv0;
+    uniform samplerCube blitTexture;
+    uniform mat4 invViewProj;
+    void main(void) {
+        vec4 projPos = vec4(uv0 * 2.0 - 1.0, 0.5, 1.0);
+        vec4 worldPos = invViewProj * projPos;
+        gl_FragColor = textureCube(blitTexture, worldPos.xyz);
+    }`;
```

Comment:
Personally I wouldn't indent here...

Reply:
I find it a lot easier to scan the code and see start / end

Reply:
Is that so?

https://github.com/playcanvas/engine/blob/main/src/platform/graphics/webgl/webgl-graphics-device.js#L46-L84

😉

Reply:
I'll add the tag, thanks. I'll also indent those shaders you mentioned :)
See MorphInstance for a good example.